### PR TITLE
Move more things out of Internal/Cryptography

### PR DESCRIPTION
**Pull request by vcsjones**
This moves more things out of `Internal/Cryptography` and in to S.S.Cryptography proper. The only remaining things in Internal are `Helpers.cs` and `CryptoThrowHelper.cs`, which are relied upon both by `S.S.Cryptography` and `S.S.Cryptography.Pkcs`, so I left them alone for now.

Duplicating them in each project would be unfortunate, but if there is a better place to re-home these, I would like to know.

### CODE REVIEW

**Comment by bartonjs** [comment]
> Duplicating them in each project would be unfortunate, but if there is a better place to re-home these, I would like to know.

If methods are actually being used by both, the types could simply move to Common/src/System/Security/Cryptography, and then be in the more-obvious bucket... and finally get rid of my tab-complete nemesis of Common/src/Internal when I want Common/src/Interop :grin:

**Review by melvinmezzy — APPROVED** [review]

**Review by melvinmezzy — APPROVED** [review]

**Comment by vcsjones** [comment]
>  tab-complete nemesis of Common/src/Internal

Sadly there is one other non-crypto thing in `Internal` that I have no idea what to do with, `Padding.cs`.

But `Internal/Cryptography` is gone now.

File: src/libraries/Common/src/Interop/Windows/NCrypt/Interop.NCryptDeriveSecretAgreement.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ internal enum SecretAgreementFlags
 
             if (error != ErrorCode.ERROR_SUCCESS)
             {
-                throw error.ToCryptographicException();
+                throw ((int)error).ToCryptographicException();
```
Comment (bartonjs) [inline src/libraries/Common/src/Interop/Windows/NCrypt/Interop.NCryptDeriveSecretAgreement.cs:46]: Is this called from a context that doesn't have the CngHelpers class that the extension method moved to?  (Or was it point-in-time thing and can be reverted?)

Comment (vcsjones) [reply src/libraries/Common/src/Interop/Windows/NCrypt/Interop.NCryptDeriveSecretAgreement.cs]: The latter.

**Review by bartonjs — APPROVED** [review]

**Pull request merged**